src/services/request-service.ts

```diff
@@ -0,0 +1,75 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
```

Comment:
I actually just found something that will work for my use case, but not all of our use cases.  If I set the id to be a concatenation of the userID and the carpoolID, as shown below, then conflict behaves as expected because it is dependent on the id field.  This works fine for my use case, but would require us to somehow predictably generate ids for all of our tables , which is not ideal.

            r.db("froyo").table("requests").insert({id: "123123", userID: "123",carpoolID: "123"},{conflict:"error"});